chapters/inheritance.tex

```diff
@@ -1041,3 +1043,131 @@ \subsection{Annotations for Redeclaration and Modification}\label{annotation-cho
+equation
+  connect(plant.y, filter.u);
+  connect(filter.y, controller.u_m);
+end FilterMeasurement;
+
+model SampledControllerSystem "Replacing non-replaceable"
+  extends System(break controller);
+  SampledController controller;
+equation
+  connect(reference.y, controller.u1); // Note: Different name
+  connect(plant.y, controller.u_m);
+  connect(controller.y, plant.u);
+end FilterMeasurement;
+
+model NewPlant "Reusable model from example"
+  extends System(break controller, break reference);
+  RealInput u;
+  RealOutput y;
+equation
+  connect(u, plant.u);
+  connect(plant.y, y);
+end NewPlant;
+\end{lstlisting}
+In these examples it would be possible to modify the \lstinline!System! model instead, but in many cases that is not realistic.
+For instance, it may not be possible to modify the \lstinline!System! and the controlled system may be comprised of a large number of components in \lstinline!System! -- instead of only two.
+\end{example}
+
+\begin{nonnormative}
+Some consequences of the rules are listed below:
+
+The syntax ensures that nested components cannot be deselected.
+
+Deselected components cannot be modified, neither in the \lstinline!extends!-clause nor when using \lstinline!C!.
+However, \lstinline!C! may add a component with same name as a deselected component (directly or through another \lstinline!extends-clause!) and that new component can be modified when using \lstinline!C!.
+
+A class using selective model extension is not necessarily a sub-type of its base class.
+
+Deselection is designed to be light-weight in particular:
+\begin{itemize}
+\item Deselection is independent of any modification.
+\item What is deselected can be determined without considering any modifications, neither of the extending class \lstinline!C! nor its base class \lstinline!B!.
+\item There is no need to instantiate any classes to know that some component is deselected (i.e., not there) for every possible instance of the model with the deselection.
+An instance tree is not required.
+\item Selective model extension operates on the syntactic level only.
+\item Conditional components can be deselected without evaluating whether they are disabled or not.
+In particular deselecting a disabled conditional component is not an error.
+Connections involving the deselected conditional component are by the deselection removed as for a disabled component.
+\item Assuming the deselections are semantically valid they can be handled in any order.
+Handling component deselections before connection deselections is only necessary to semantically check that a connection deselection does not involve a deselected component.
+\end{itemize}
+
+\begin{example}
+The syntactic equivalence of connection deselection ensures that connect-statements in for-loops can be deselected:
+\begin{lstlisting}[language=modelica]
+model B
+  $\ldots$
+equation
+  if b then
+    for i in 2:10 loop
+      connect( // Let's add some whitespace for fun.
+        a[i],
```

Comment:
When looking through the new specification text I was thrown off by this and the two white space comments. What is the significance of these? Removing those seems much less confusing to the reader.

Reply:
The point is that the equations are not compared for textual equality, implying that white-space and comments will not destroy equivalence.

Reply:
OK, in that case, I would probably remove " for fun" which in this specification gives a non-professional impression. I first thought it was some kind of intermediate test that was forgotten to clean up.

Reply:
OK, let's restart this with a conversation attached to the line to be changed.  Resolving.